hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SQLServerParameters.java

```diff
@@ -34,24 +36,48 @@ public String process(String sql, int parameterCount) {
      */
     @Override
     public String processLimit(String sql, Object[] parameterArray, boolean hasOffset) {
-        if (isProcessingNotRequired(sql)) {
+        if ( isProcessingNotRequired( sql ) ) {
             return sql;
         }
 
         // Replace 'offset ? fetch next ? rows only' with the @P style parameters for Sql Server
         int index = hasOffset ? parameterArray.length - 1 : parameterArray.length;
-        int pos = sql.indexOf( " offset ?" );
+        // See SQLServer2012LimitHandler#getOffsetFetch
+        String offsetQueryString;
+        int pos = sql.indexOf( " offset ? rows fetch next ? rows only" );
+        if ( pos > -1 ) {
+            offsetQueryString = " offset @P" + index++;
+        }
+        else {
+            pos = sql.indexOf( " offset 0 rows fetch next ? rows only" );
+            offsetQueryString = " offset 0";
+        }
```

Comment:
@DavideD wouldn't it be good enough to just match `" offset "` and then grab the next two characters?

Reply:
Nice idea.
It bothers me a bit that we check such a short string but I've updated the code anyway